### DISCUSSION TITLE
Testing Notifications System with Jest / React Testing Library, NotificationsProvider is undefined?

### DISCUSSION THREAD

**Discussion by ridgekuhn** [mantinedev/mantine#3392] [Q&A] [answered]
Hello, I'm trying to test the [Notifications System](https://mantine.dev/others/notifications/) package with Jest and React Testing Library, but I'm having trouble rendering a component with the context providers. I get the following error from React-DOM:

"Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports."

The test looks something like this:

```javascript
import { MantineProvider } from "@mantine/core";
import { NotificationsProvider } from "@mantine/notifications";
import { render } from "@testing-library/react";
import MyComponent from "@/src/my-component.jsx";

test("Show notification", async () => {
  const component = render(
    <MantineProvider>
      <NotificationsProvider>
        <MyComponent />
      </NotificationsProvider>
    </MantineProvider>
  );
});
```

I've also tried just rendering the entire app component (NextJS 12, if that makes a difference) with the test component as a child prop, but I got the same error. 

Interestingly, if I `console.log(NotificationsProvider)` either in the app component file where I actually use the NotificationsProvider, or in the test file as above, `NotificationsProvider` is always `undefined`. Am I doing something wrong?

Thanks in advance!

**Comment by rtivital** [comment, marked as answer]
I suppose that your testing environment cannot recognize `import MyComponent from "@/src/my-component.jsx";` import.
Mantine packages export components correctly, otherwise they would not work.

**Reply by ridgekuhn** [reply]
@rtivital Thanks for taking the time to respond. Sorry for the confusion, that's just an example import path, I have no problem resolving the imports for components in the affected tests, it's only when I pass them to render() wrapped with MantineProvider and NotificationsProvider that I get the error.

**Reply by ridgekuhn** [reply]
@rtivital So sorry, please ignore this thread. The issue was caused by a leftover mock file overwriting the import! 😅
